bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDCyclicTransposeFlowGraphFinder.java

```diff
@@ -0,0 +1,184 @@
+
+            List<String> previousNodesInTransposeFlow = new ArrayList<>();
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getDestinationNode()
+                    .getEntityName());
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName());
+
+            List<DFDVertex> finalVertices = vertices;
+            vertices = (incomingFlowsToPin.stream()
+                    .flatMap(flow -> handleIncomingFlow(flow, inputPin, finalVertices, sourceNodes, previousNodesInTransposeFlow).stream())
+                    .toList());
+        }
+        return vertices;
+    }
+
+    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> inputPins, List<Node> sourceNodes, List<String> previousNodesInTransposeFlow) {
+        List<DFDVertex> vertices = new ArrayList<>();
+        vertices.add(sink);
+
+        if (sourceNodes.contains(sink.getReferencedElement())) {
+            return vertices;
+        }
+
+        for (Pin inputPin : inputPins) {
+            List<Flow> incomingFlowsToPin = dataFlowDiagram.getFlows()
+                    .stream()
+                    .filter(flow -> flow.getDestinationPin()
+                            .equals(inputPin))
+                    .toList();
+
+            if (!LoopCheck(previousNodesInTransposeFlow, incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName()))
+                continue;
+
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName());
+
+            List<DFDVertex> finalVertices = vertices;
+            vertices = (incomingFlowsToPin.stream()
+                    .flatMap(flow -> handleIncomingFlow(flow, inputPin, finalVertices, sourceNodes, previousNodesInTransposeFlow).stream())
+                    .toList());
+        }
+        return vertices;
+    }
+
+    public List<DFDVertex> handleIncomingFlow(Flow incomingFlow, Pin inputPin, List<DFDVertex> vertices, List<Node> sourceNodes,
+            List<String> previousNodesInTransposeFlow) {
+        List<DFDVertex> result = new ArrayList<>();
+
+        Node previousNode = incomingFlow.getSourceNode();
+
+        List<Pin> previousNodeInputPins = getAllPreviousNodeInputPins(previousNode, incomingFlow);
+
+        // Das new DFDVertex & Flow Copy um Problem zu behandeln.
+        List<DFDVertex> previousNodeVertices = determineSinks(new DFDVertex(previousNode, new HashMap<>(), new HashMap<>()), previousNodeInputPins,
+                sourceNodes, new ArrayList<>(previousNodesInTransposeFlow));
+
+        for (DFDVertex vertex : vertices) {
+            result.addAll(cloneFlowandVertexForMultipleFlowGraphs(vertex, inputPin, incomingFlow, previousNodeVertices));
+        }
+        return result;
+    }
+
+    private Boolean LoopCheck(List<String> list, String element) {
```

Comment:
Names of Methods should be in camel case?
Parameters should be named somewhat expressively

Reply:
changed to loopCheck(List<String> previousNodesInTransposeFlow, String sourceNode)